packages/client-utils/src/sync-tools.js

```diff
@@ -48,19 +49,50 @@ export const retryUntilCondition = async (
   operation,
   condition,
   message,
-  { maxRetries = 6, retryIntervalMs = 3500, log = console.log, setTimeout },
+  {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    reusePromise,
```

Comment:
consider for clarity
```suggestion
    reusePromise = false,
```